tests/unit/general/test_mem.cpp

```diff
@@ -12,11 +12,68 @@
 #include "mfem.hpp"
 #include "unit_tests.hpp"
 
+using namespace mfem;
+
+TEST_CASE("MemoryManager/Scopes",
+          "[MemoryManager]"
+          "[CUDA]")
+{
+   SECTION("WithNewMemoryAndSize")
+   {
+      Vector x(1);
+      x.UseDevice(true);
+      {
+         Vector X;
+         // from Operator::InitTVectors
+         X.NewMemoryAndSize(x.GetMemory(), x.Size(), false);
+         // from Vector::SetSubVectorComplement
+         X.Read();
+         // from Operator::RecoverFEMSolution
+         x.SyncMemory(X);
+      }
+      // Accessible Memory<double> to get the flags
+      struct MemoryDouble
+      {
+         double *h_ptr;
+         int capacity; ///< Size of the allocated memory
+         MemoryType h_mt; ///< Host memory type
+         mutable unsigned flags;
+      };
+      const MemoryDouble *mem = (MemoryDouble*) &x.GetMemory();
+      const double *h_x = mem->h_ptr;
+      REQUIRE(h_x == x.GetData());
+      REQUIRE(mem->capacity == x.Size());
+      REQUIRE(mem->h_mt == Device::GetHostMemoryType());
+      constexpr unsigned REGISTERED = 1 << 0;
+      const bool registered = mem->flags & REGISTERED;
+      const bool registered_is_known = registered == mm.IsKnown(h_x);
+      // Failing CUDA test with NewMemoryAndSize/Read/SyncMemory
+      if (Device::IsEnabled()) { REQUIRE_FALSE(registered_is_known); }
+      else { REQUIRE(registered_is_known); }
+   }
+
+   SECTION("WithMakeRef")
+   {
+      Vector x(1);
+      x.UseDevice(true);
+      const double *x_data = x.GetData();
+      {
+         Vector X;
+         // from Operator::InitTVectors
+         X.MakeRef(x, 0, x.Size());
+         // from Vector::SetSubVectorComplement
+         X.Read();
+         // from Operator::RecoverFEMSolution
+         x.SyncMemory(X);
```

Comment:
I think you're supposed to use the following since you used `MakeRef` (and not `NewMemoryAndSize(...,...,false)`):
```suggestion
         x.SyncAliasMemory(X);
```

Reply:
You're probably true. The idea here is to show what is called through `BilinearForm::RecoverFEMSolution`, which should open the discussion.